consumer/consumer.go

```diff
@@ -88,7 +127,9 @@ func (c *Consumer) Serve(clientID string, addrs ...string) error {
 			// annoying little issue.
 			err = errors.Wrap(err, "__consumer_offsets topic doesn't yet exist, either because no client has yet requested an offset, or because this consumer group is not yet functioning at startup or after rebalancing.")
 		}
-		return errors.Wrap(err, fmt.Sprintf("failed to create a consumer for topics %+v in consumer group %q", topics, consumerGroup))
+		err = errors.Wrap(err, fmt.Sprintf("failed to create a consumer for topics %+v in consumer group %q", topics, consumerGroup))
+		common.Logger.Println(err)
```

Comment:
We should not log errors if we return them, this might generate a lot of lines for only one error.
We should return errors without logging and it's the caller's responsibility to log or to bubble up the error.

> If you choose to handle the error by logging it, by definition it’s not an error any more — you handled it.

-- Dave Cheney (again)

Reply:
Yes, actually I strongly agree - initially we removed logging completely from the package because I didn't really see anything that needed logging.  ;-)